core/internal/filetransfer/file_transfer_default.go

```diff
@@ -108,8 +108,13 @@ func (ft *DefaultFileTransfer) Upload(task *Task) error {
 	resp, err := ft.client.Do(req)
 	if err != nil {
 		return err
-	} else if resp == nil || resp.StatusCode < 200 || resp.StatusCode > 299 {
-		return fmt.Errorf("file transfer: upload: failed to upload: %v", resp.Status)
+	}
+	// TODO: why would we get an empty response?
```

Comment:
I have to consider it a bug in the hashicorp client...

Reply:
yeah could be need to investigate this, i'll open a ticket!